File: MapA7/MapA7/Model/Statements/IfStatement.cs
using MapA7.Model.Expressions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model.Statements
{
    public class IfStatement:IStatement
    {
        private IExpression expression;
        private IStatement sthen;
        private IStatement selse;

        public IfStatement(IExpression expression, IStatement sthen, IStatement selse)
        {
            this.expression = expression;
            this.sthen = sthen;
            this.selse = selse;
        }

        public ProgramState Execute(ProgramState ps)
        {
            SymbolTable symbolTable = ps.GetSymbolTable();
            ExecutionStack executionStack = ps.GetExecutionStack();
            int result = expression.Evaluate(symbolTable);
            if (result == 0) executionStack.Push(selse);
            else executionStack.Push(sthen);
            return ps;
        }

        public String Tostring() { return "if " + expression.Tostring() + " then " + sthen.Tostring() + " else " + selse.Tostring(); }
    }
}

File: MapGUI3/src/Model/Statements/IfStatement.java
package Model.Statements;

import Model.ExecutionStack;
import Model.ProgramState;
import Model.SymbolTable;
import Model.expressions.Expression;

public class IfStatement implements IStatement {
    private Expression expr;
    private IStatement thenStatement;
    private IStatement elseStatement;

    public IfStatement(Expression expr, IStatement thenStatement, IStatement elseStatement) {
        this.expr = expr;
        this.thenStatement = thenStatement;
        this.elseStatement = elseStatement;
    }

    public ProgramState execute(ProgramState ps) {
        SymbolTable<String, Integer> symTable = ps.getSymTable();
        ExecutionStack<IStatement> executionStack = ps.getExecutionStack();
        int result = expr.evaluate(symTable,ps.getHeap());
        if (result == 0) executionStack.push(elseStatement);
        else executionStack.push(thenStatement);
        return ps;
    }

    public String toString() {
        return "If " + expr.toString() + " Then " + thenStatement.toString() + " Else " + elseStatement.toString();
    }

}

